pkg/apis/apps/types.go

```diff
@@ -82,6 +97,16 @@ type StatefulSetSpec struct {
 	// pattern: pod-specific-string.serviceName.default.svc.cluster.local
 	// where "pod-specific-string" is managed by the StatefulSet controller.
 	ServiceName string
+
+	// PodCreationPolicy controls how pods are created during initial scale up,
+	// when replacing pods on nodes, or when scaling down. The default policy is
+	// `OrderedReady`, where pods are created in increasing order (pod-0, then
+	// pod-1, etc) and the controller will wait until each pod is ready before
+	// continuing. When scaling down, the pods are removed in the opposite order.
+	// The alternative policy is `Parallel` which will create pods in parallel
+	// to match the desired scale without waiting, and on scale down will delete
+	// all pods at once.
+	PodCreationPolicy PodCreationPolicyType
```

Comment:
When updates are in place, the stateful set controller should own the availability of the set, and replacement policy during an update is covered by the update strategy.  So like deployment, it is an override (which makes sense since it shows up as a sub field)